src/clients/customer/lib/ui/home/home_view.dart

```diff
@@ -9,7 +10,35 @@ class HomeView extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return ViewModelBuilder<HomeViewModel>.reactive(
-      builder: (context, model, child) => Scaffold(),
+      builder: (context, model, child) => Scaffold(
+          body: model.isBusy
+              ? Center(
+                  child: CircularProgressIndicator(),
+                )
+              : model.hasError
+                  ? BoxText.headingThree(
+                      'An error has occered while running the future',
+                    )
+                  : model.data.isEmpty
+                      ? Text('There is currently no merchants for this region')
+                      : ListView.builder(
+                          padding: EdgeInsets.symmetric(
+                              vertical: screenHeightPercentage(context,
+                                  percentage: 0.1),
+                              horizontal: screenHorizontalPadding),
+                          itemCount: model.data.length,
+                          itemBuilder: (context, index) => Padding(
+                                padding: const EdgeInsets.only(bottom: 24),
+                                child: LargeMerchantItem(
+                                  images: model.data[index].images ?? [],
```

Comment:
Instead of repeating the `model.data[index]` code we can store it as `final merchantItem = model.data[index]` at the beginning of the `itemBuilder` function. This will make it easier to teach when explaining.